modules/core/04-channel/keeper/packet.go

```diff
@@ -128,15 +128,23 @@ func (k Keeper) RecvPacket(
 		return errorsmod.Wrap(types.ErrChannelNotFound, packet.GetDestChannel())
 	}
 
-	if !collections.Contains(channel.State, []types.State{types.OPEN, types.TRYUPGRADE, types.ACKUPGRADE}) {
-		return errorsmod.Wrapf(types.ErrInvalidChannelState, "expected channel state to be one of [%s, %s, %s], but got %s", types.OPEN.String(), types.TRYUPGRADE.String(), types.ACKUPGRADE.String(), channel.State.String())
+	if !collections.Contains(channel.State, []types.State{types.OPEN, types.STATE_FLUSHING}) {
+		return errorsmod.Wrapf(types.ErrInvalidChannelState, "expected channel state to be one of [%s, %s], but got %s", types.OPEN, types.STATE_FLUSHING, channel.State)
 	}
 
-	// in the case of the channel being in TRYUPGRADE or ACKUPGRADE we need to ensure that the channel is not in flushing,
-	// and that the counterparty last sequence send is less than or equal to the packet sequence.
-	if counterpartyLastSequenceSend, found := k.GetCounterpartyLastPacketSequence(ctx, packet.GetDestPort(), packet.GetDestChannel()); found {
-		if channel.FlushStatus != types.FLUSHING || packet.GetSequence() > counterpartyLastSequenceSend {
-			return errorsmod.Wrapf(types.ErrInvalidFlushStatus, "expected channel flush status to be (%s) when counterparty last sequence send (%d) is set, failed to recv packet (%d)", types.FLUSHING, counterpartyLastSequenceSend, packet.GetSequence())
+	// in the case of the channel being in FLUSHING we need to ensure that the the counterparty last sequence send
+	// is less than or equal to the packet sequence.
+	if channel.State == types.STATE_FLUSHING {
+		counterpartyUpgrade, found := k.GetCounterpartyUpgrade(ctx, packet.GetDestPort(), packet.GetDestChannel())
+		if !found {
+			return errorsmod.Wrapf(types.ErrUpgradeNotFound, "counterparty upgrade not found for channel: %s", packet.GetDestChannel())
+		}
+
+		if packet.GetSequence() > counterpartyUpgrade.LatestSequenceSend {
```

Comment:
There is one gotcha here! 

After TRY handler executes, we have `counterpartyUpgrade` and are in `FLUSHING` channel state **BUT** `LastSequenceSend` is not set yet, until after `ACK` and `CONFIRM` succeed. 
Right now, that means relayer will have `recvPacket` rejected until it goes further in the upgrade handshake and then it would need to retry packets. 

I wonder if wrapping this check in:
```go
if counterpartyUpgrade.LastSequenceSend != 0 { 
    // do greater than checking 
}
```

This way we know that if `LastSequenceSend` is still 0 that we can safely continue to receive packets while in `FLUSHING` state.

Happy to leave this as a future improvement, I'll open an issue based off this comment.

Reply:
lol, damn nice catch. Can pick it up after this PR is merged, along with the recv test refactor.

Reply:
I added the needs discussion label to that one and put it into the beta milestone since its not really the spec behaviour

Reply:
sgtm